src/utils/multiformats/src/stack_string.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+// A stack-allocated string to avoid allocations
+pub struct StackString<const S: usize> {
+    buf: [u8; S],
+    len: usize,
+}
+
+impl<const S: usize> StackString<S> {
+    pub fn new(buf: [u8; S], len: usize) -> Self {
```

Comment:
just for my understanding:
**Q:** do we always have a stack allocated buffer here? How is it checked? Is that granted by type declaration (array decl)?

Reply:
The idea is that higher level code that uses `StackString` has to know the max buffer size of the string its producing and pass it as const generic param.

Not fully sure what you mean by "How is it checked?". By accepting `buf` as array `[u8; S]` and not as slice `&[u8]` we always know the exact size at compile time, i.e. it's impossible to pass `[0; 10]` to a `StackString<11>`.

Reply:
Note that this type is likely temporary. If we like the ergonomics of it and will want to extend the use of stack-allocated strings to other parts (like dataset names, aliases, references) - we will probably switch to [compact_str](https://lib.rs/crates/compact_str).

Reply:
> By accepting `buf` as array `[u8; S]` and not as slice `&[u8]` we always know the exact size at compile time, i.e. it's impossible to pass `[0; 10]` to a `StackString<11>`.

yep, thanks for the clarification -- I got the idea right